homeassistant/components/meraki/device_tracker.py

```diff
@@ -43,8 +48,22 @@ def __init__(self, config, async_see):
         self.validator = config[CONF_VALIDATOR]
         self.secret = config[CONF_SECRET]
 
+    def validate_secret(self, secret):
+        """Validate shared secret."""
+        if not secret:
+            _LOGGER.error("no secret supplied")
+            return False
+        if secret != self.secret:
+            _LOGGER.error("invalid secret supplied")
+            _LOGGER.debug("received secret %s", secret)
```

Comment:
We shouldn't log secrets.
```suggestion
            _LOGGER.error("Invalid secret supplied")
```